aptos-move/e2e-tests/src/executor.rs

```diff
@@ -148,6 +158,11 @@ pub struct FakeExecutorImpl<O: OutputLogger> {
     /// s.t. the comparison test is executed (BothComparison).
     executor_mode: Option<ExecutorMode>,
     allow_block_executor_fallback: bool,
+    /// When set (only in fuzzing/test constructor with external thread pool), reuse a shared
+    /// module cache manager across transaction blocks to enable cache persistence.
+    shared_module_cache_manager: Option<AptosModuleCacheManager>,
+    /// Monotonic counter for generating consecutive block metadata when using a shared cache.
+    next_block_id: AtomicU64,
```

Comment:
Why this has to be an atomic? I think encapsulating it into some structure that own manager is better, as I am not sure fake executor provides guarantees on this incremented and it is purely for fuzzing